### DISCUSSION TITLE
How do I go about bringing burn-rs onto an embedded device? (microcontroller)

### DISCUSSION THREAD

**Discussion by exo-cortex** [tracel-ai/burn#1911] [General]
Hello, 
I would like to try out a minimalist example that can run on a microcontroller board. My plan is more like a proof of concept and It could theoretically be incredible small / simple. I read on the website that burn supports `no_std`, but I haven't seen any examples yet. I would like to get a general direction or suggestions on which approach I should take for this project. 
I would like to use it on a raspberry pi pico or an arduino (maybe not the old UNO, but who knows). 

Thank you very much

**Comment by laggui** [comment]
We don't have an in-depth tutorial yet but we're looking to expand on that soon! Take a look at the discussion in [this issue](https://github.com/tracel-ai/burn/issues/1067) just in case.

I don't have any of these devices yet but if you want to link your open source project it would be helpful to get an official example/tutorial started! 

And feel free to ask away on discord, I know some other users have been down this road in the past.

**Reply by exo-cortex** [reply]
thank you!
I did search for similar topics, but somehow missed that one. Sorry about that!

**Reply by laggui** [reply]
No worries!

**Comment by antimora** [comment]
Just be aware of these two existing issues:

1. #1067
2. #302
